### PR TITLE
Using overwrite rather than deprecated cobber of astropy.io.fits

### PR DESCRIPTION
**Pull request by bsipocz**
`clobber` in astropy has been deprecated for a while now in favour of `overwrite` and thus issues a DeprecationWarnings that makes the CI fail for downstream packages.

### CODE REVIEW

**Comment by DougBurke** [comment]
@bsipocz do you know what version of astropy added the `overwrite` jeyword argument (i.e. what is the minimum version of astropy needed after this change)?

**Comment by bsipocz** [comment]
I can double check, but v1.3 already had it I believe (and the deprecation warning was added in 2.0).

**Comment by DougBurke** [comment]
Can you comment out (or delete) the two warnings related at https://github.com/sherpa/sherpa/blob/master/sherpa/conftest.py#L107 (and the following line) since your change should mean these can be removed?

**Comment by DougBurke** [comment]
We don't mention a minimum version of astropy in our `README` file. Do you think we need to @bsipocz ? There's also the fact we don't test against too many versions of astropy (I forget what versions we use in our travis config), so we might have to say something like "The minimum version is 1.3 but testing is only done with 2.x" (or whatever we use).

**Comment by bsipocz** [comment]
@DougBurke - Since as I see sherpa only uses `astropy.io.fits`, it's safe to just use different versions in each jobs you already have on travis, so then it's easier to say that the minimum is 1.3.

Should I update the travis file here to check whether that would work?

**Comment by DougBurke** [comment]
I'm reluctant to change the current set of versions being tested on travis,
as I know @olaurino has had to pick them carefully in the past (e.g. for
our XSPEC tests). So I'll leave that decision to him (we are also trying to
not run too many tests since we are using the free service provided by
Travis).

**Comment by olaurino** [comment]
The oldest version of astropy we test against is 2.0.x. So far I have been relying on the fact that we use only `astropy.io.fits` and that this is supposed to be backwards compatible with `pyfits`. In other terms I have not been too concerned with making sure we test against all the possible versions of astropy. If you feel we should add a test against 1.3 that is fine with me, otherwise I'd leave things as they are but add some text in the README to say that the minimum required version is now 1.3. And as suggested by Doug we might add that however we only test against 2.x and 3.x.

**Comment by olaurino** [comment]
Perfect, thanks!

File: sherpa/astro/io/pyfits_backend.py [hunk 1]
```diff
@@ -1010,7 +1010,7 @@ def set_table_data(filename, data, col_names, hdr=None, hdrnames=None,
     tbl.name = 'HISTOGRAM'
     if packup:
         return tbl
-    tbl.writeto(filename, clobber=True)
+    tbl.writeto(filename, overwrite=True)
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply sherpa/astro/io/pyfits_backend.py]: Of course it should. Thanks for noticing.
A separate question of course is that you may want to consider deprecate clobber in favour of overwrite, but that's well beyond the scope of this PR.

**Comment by DougBurke** [comment]
@olaurino are we going to pull this in for the next release?

**Comment by DougBurke** [comment]
I'd mentioned that we should mention it in the `README.md` file, but that's something we can add.

**Pull request merged**